### DISCUSSION TITLE
How can I get the context (auth user) outside of GraphQL resolvers

### DISCUSSION THREAD

**Discussion by Drakota** [apollographql/apollo-server#4404] [General]
Hi, I need to get the currently authenticated user (from a JWT passed in a authorization header), outside of my GraphQL resolvers.

I have something like this:
```ts
users: async (parent, args, ctx, info): Promise<User[]> => {
    // Some logic
    const permissions = await Permissions.findAll();
   // Easy fix const permissions = await Permissions.findAll(ctx);
}

// Permissions.ts
class Permission  {
    public static async findAll(): Promise<Permission[]> {
        // I need the auth user here
    } 
}
```
I've overridden my ORMs findAll method to do some authorization checks to fetch only specific permissions by taking in consideration the authenticated user. The thing is now I'm outside of my resolver, but want to access the user. An easy fix would just to pass the context to my function and be done with it, but I have other use cases where I would find it beneficial to simply be able to get the authenticated user instead of passing the context down everytime.

I've looked at this part in the docs about [authorization outside of graphql](https://www.apollographql.com/docs/apollo-server/security/authentication/#authorization-outside-of-graphql), but in my case it's not the be used inside a REST endpoint.

**Comment by FezVrasta** [comment]
You need to pass the context around, there is no other way around it.
